org.eclipse.jface.text/src/org/eclipse/jface/text/contentassist/AsyncCompletionProposalPopup.java

```diff
@@ -374,19 +374,19 @@ protected List<CompletableFuture<List<ICompletionProposal>>> buildCompletionFutu
 			return Collections.emptyList();
 		}
 		List<CompletableFuture<List<ICompletionProposal>>> futures = new ArrayList<>(processors.size());
-		for (IContentAssistProcessor processor : processors) {
-			// Use a custom ForkJoinWorkerThreadFactory, to prevent issues with a
-			// potential SecurityManager. Threads created by ForkJoinPool.commonPool(),
-			// which is used in CompletableFuture.supplyAsync(), get no permissions.
-			ForkJoinPool commonPool= new ForkJoinPool(ForkJoinPool.getCommonPoolParallelism(),
-					pool -> new ForkJoinWorkerThread(pool) {
-						// anonymous subclass to access protected constructor
-					}, null, false);
-			try {
+		// Use a custom ForkJoinWorkerThreadFactory, to prevent issues with a
+		// potential SecurityManager. Threads created by ForkJoinPool.commonPool(),
+		// which is used in CompletableFuture.supplyAsync(), get no permissions.
+		ForkJoinPool commonPool= new ForkJoinPool(ForkJoinPool.getCommonPoolParallelism(),
```

Comment:
This still makes a new pool bound to the `invocationOffset`, e.g., for one invocation of content assist.

Please see this remark, which holds true here, too: https://github.com/eclipse-jdt/eclipse.jdt.core/pull/1017#issuecomment-1534323526

Reply:
This seems to be a bigger change. I tend to revert #185 to prevent the creation of new pools for every code completion and maybe we find a proper solution for the issue.

Reply:
> This seems to be a bigger change. I tend to revert #185 to prevent the creation of new pools for every code completion and maybe we find a proper solution for the issue.

Yes, please.